### PR TITLE
PERF: stop using `RsModuleIndex` in `RsFile.doGetCachedData`

### PR DESCRIPTION
**Pull request by vlad20012**
Slightly speeds up name resolution. This is also a proper fix for #9110. 

changelog: Slightly speed up name resolution

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt [hunk 1]
```diff
@@ -695,15 +696,29 @@ private fun ModData.toRsModNullable(project: Project): List<RsMod> {
...
 fun findModDataFor(file: RsFile): ModData? {
     val project = file.project
     val defMapService = project.defMapService
-    val virtualFile = file.virtualFile as? VirtualFileWithId ?: return null
-    // TODO Ensure def maps are up-to-date (`findCrates` may return old crate of def maps haven't updated).
+    val virtualFile = file.virtualFile ?: return null
+    if (virtualFile !is VirtualFileWithId) return null
+
+    if (!defMapService.areAllDefMapsUpToDate()) {
+        // Ensure def maps are up-to-date (`findCrates` may return old crate of def maps haven't updated)
```
Comment (dima74) [inline src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt:706]:
```suggestion
        // Ensure def maps are up-to-date (`findCrates` may return old crate if def maps haven't updated)
```

**Review by dima74 — COMMENTED** [review]
Should we also update `RsFile.isIncludedByIncludeMacro`?

**Comment by vlad20012** [comment]
> Should we also update `RsFile.isIncludedByIncludeMacro`?

I'll send a separate PR

**Review by dima74 — APPROVED** [review]
bors r+

**Comment by vlad20012** [comment]
bors r=dima74

**Pull request merged**